### PR TITLE
Noviny/apply release plan

### PR DESCRIPTION
**Pull request by Noviny**
No tests + JANK code, hasn't been run yet. Can I get a Work in Progress up in this PR?

assuming I wrote perfect code and there are no bugs, this is done 😇

### CODE REVIEW

File: packages/apply-release-plan/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+  // iterate over releases updating packages
+  let finalisedRelease = await Promise.all(
+    releaseWithChangelogs.map(release => {
+      return versionPackage(release, versionsToUpdate);
+    })
+  );
+
+  let prettierConfig = await prettier.resolveConfig(cwd);
+
+  for (let release of finalisedRelease) {
```
Comment (mitchellhamilton) [inline packages/apply-release-plan/src/index.ts]: Not something we need to do now but post-v2, let's make this happen in parallel and only write after we've got all the new changelog entries #21

File: packages/apply-release-plan/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+function getNewChangelogEntry(
+  releaseWithWorkspaces: ModCompWithWorkspace[],
+  changesets: NewChangeset[],
+  changelogConfig: false | readonly [string, any],
+  cwd: string
+) {
+  let getChangelogFunc: ChangelogFunction = () => Promise.resolve("");
+  let changelogOpts: any;
+  if (changelogConfig) {
+    let changelogPath = resolveFrom(cwd, changelogConfig[0]);
```
Comment (mitchellhamilton) [inline packages/apply-release-plan/src/index.ts]: This should resolve like this
```suggestion
    let changelogPath = resolveFrom(path.join(cwd, ".changeset"), changelogConfig[0]);
```

**Comment by Noviny** [comment]
> Is packages/apply-release-plan/src/test-utils/get-changelog-dep-updated.ts used at all, it shows as an empty file on GitHub?

It goes along with a bunch of skipped tests so I could get this 'done enough' - deliberately added so its discrepancy is obvious, even though it's an empty file.

**Comment by Noviny** [comment]
> only write after we've got all the new changelog entries

This bit's already implemented but not tested (we call getNewChangelogEntry before we start writing, `finalisedRelease` which we're iterating over here has the written changelogs. #21 should be closed when this is implemented.

Parallelising writing makes sense for later.

**Pull request merged**